pkg/kinflate/commands/diff_test.go

```diff
@@ -89,9 +90,18 @@ func TestDiff(t *testing.T) {
 			}
 			buf := bytes.NewBuffer([]byte{})
 			err = diffOps.RunDiff(buf, os.Stderr, fs)
-			if err != nil {
+			switch {
+			case err != nil && len(testcase.ExpectedError) == 0:
 				t.Errorf("unexpected error: %v", err)
+			case err != nil && len(testcase.ExpectedError) != 0:
+				if !strings.Contains(err.Error(), testcase.ExpectedError) {
```

Comment:
I don't think it's super good practice to grep on error strings :-/

PS: Same below

Reply:
There are 2 reasons that I choose not to use it.
- the error message doesn't make sense for us: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go#L88-L91
- all test config  are loaded from a yaml config file. IOW it doesn't know about the error type.